### PR TITLE
Partial CMS configuration

### PR DESCRIPTION
**Pull request by jrbl**
- Needs more work, but seems to have many of the basics.

This is at least part of a CMS setup. I'd love to have @jarv and @jbau look this over, or possibly pull it into a local branch and finish the job.

### CODE REVIEW

File: playbooks/roles/cms/templates/auth.json.j2 [hunk 1]
```diff
@@ -0,0 +1 @@
+{{ auth_config | to_nice_json }}
```
Comment (jarv) [inline playbooks/roles/cms/templates/auth.json.j2:2]: So for the slight variations that will be in the auth and env json files for the cms I was thinking we could do something like this in the jinja template

```
 {% do auth_config.update( { < override some hash keys > } )  %}
 {{ auth_config | to_nice_json }}
```

I couldn't think of any cleaner way to do it.

File: playbooks/roles/cms/vars/main.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    'staticfiles':
+      'BACKEND':  'django.core.cache.backends.memcached.MemcachedCache'
+      'KEY_FUNCTION':  'util.memcache.safe_key'
+  'LOG_DIR': '/mnt/logs/edx'
+  'LOGGING_ENV': 'lms-dev'
+  'SITE_NAME': 'studio.lms-dev.m.edx.org'
+  'SYSLOG_SERVER': 'syslog.a.m.i4x.org'
+  'LMS_BASE': 'lms-dev.m.edx.org'
+  'SESSION_COOKIE_DOMAIN': '.lms-dev.m.edx.org'
+
```
Comment (jarv) [inline playbooks/roles/cms/vars/main.yml:29]: I'm actually not certain if roles have an isolated variable namespace unfortunately.
There was a recent thread on the ansible list that I commented on here -> https://groups.google.com/forum/?fromgroups=#!topic/ansible-project/rUPBdFP6SVc

We may just want to share the same variable for cms, lms, lms-preview and override specific keys in the jinja template.

File: playbooks/roles/lms/vars/main.yml [hunk 3]
```diff
@@ -97,11 +97,11 @@ lms_debian_pkgs:
...
-  - puppet
-  - puppet-common
-  - puppet-lint
-  - puppetmaster
-  - puppetmaster-common
+  #- puppet
+  #- puppet-common
+  #- puppet-lint
+  #- puppetmaster
+  #- puppetmaster-common
```
Comment (jarv) [inline playbooks/roles/lms/vars/main.yml]: yeah we can definitely remove these :)

File: playbooks/roles/nginx/templates/cms.j2 [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  if ($http_x_forwarded_proto = "http") {
+    set $do_redirect "true";
+  }
+
+  # Run our actual redirect...
+  if ($do_redirect = "true") {
+    rewrite ^ https://$host$request_uri? permanent;
+  }
+
+}
```
Comment (jarv) [inline playbooks/roles/nginx/templates/cms.j2:97]: Do you think you guys are going to use your own template with custom domain info or should we template this one out with variables?

File: playbooks/roles/cms/templates/auth.json.j2 (hunk 1 above)
Comment (jrbl) [inline playbooks/roles/cms/templates/auth.json.j2:2]: I think I see what you mean here, though at this point I still don't have any real idea what these values should be. :)

File: playbooks/roles/cms/vars/main.yml (hunk 2 above)
Comment (jrbl) [inline playbooks/roles/cms/vars/main.yml:29]: Yes, I think this sounds reasonable. Or maybe carrying around a dictionary or namespacing as you suggested. But between lms and cms, most of these values seem to be the same anyway.

File: playbooks/roles/lms/vars/main.yml (hunk 3 above)
Comment (jrbl) [inline playbooks/roles/lms/vars/main.yml]: This is probably trumped by @e0d's branch to skim cruft and move things around. woot.

File: playbooks/roles/nginx/templates/cms.j2 (hunk 4 above)
Comment (jrbl) [inline playbooks/roles/nginx/templates/cms.j2:97]: We'll have our own, like we do for LMS.

**Pull request closed without merging**